### ISSUE TITLE
Error after upgrading to 0.7.0 regarding MX records

### ISSUE THREAD

**Issue by ericrichtert** [akamai/terraform-provider-akamai#154]
After upgrading from provider version 0.5.0 to 0.7.0, we can't terraform plan the code anymore on a MX record.

Error:
Error: Recordset [xxxx.yyy MX]: Remote has diverged from TF Config. Manual intervention required.

### Terraform Version
λ terraform version
Terraform v0.12.24
+ provider.akamai v0.7.0
+ provider.aws v2.63.0

### Affected Resource(s)
Please list the resources as a list, for example:
- akamai_dns_record - MX

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
resource "akamai_dns_record" "mx__xxxx_yyy" {
  zone       = "xxx.yyy"
  name       = "xxx.yyy"
  active     = true
  recordtype = "MX"
  ttl        = local.default_ttl
  priority   = 10
  target = [
    "mx1."xxx.yyy"",
    "mx2."xxx.yyy""
  ]
}
```

### Debug Output
2020-05-26T12:57:45.076+0200 [DEBUG] plugin.terraform-provider-akamai_v0.7.0_x4.exe: 2020/05/26 12:57:45 [DEBUG] [Akamai DNSv2] RECORD READ [&{xxxx.yyy MX 3600 false [10 mx1.xxxx.yyy. 10 mx2.xxxx.yyy.eu.] 0 0 0 0  0 0    0 0 0     0       0     0 0 0 0  0}] [xxxx.yyy] [xxxx.yyy] [MX] 
2020-05-26T12:57:45.076+0200 [DEBUG] plugin.terraform-provider-akamai_v0.7.0_x4.exe: 2020/05/26 12:57:45 [DEBUG] [Akamai DNSv2] READ record data read JSON ["10 mx1.xxxx.yyy.eu.","10 mx2.xxxx.yyy."]
2020-05-26T12:57:45.076+0200 [DEBUG] plugin.terraform-provider-akamai_v0.7.0_x4.exe: 2020/05/26 12:57:45 MX READ. TARGET HAS NOT CHANGED
2020/05/26 12:57:45 [ERROR] <root>: eval: *terraform.EvalRefresh, err: Recordset [xxxx.yyy MX]: Remote has diverged from TF Config. Manual intervention required.
2020/05/26 12:57:45 [ERROR] <root>: eval: *terraform.EvalSequence, err: Recordset [xxxx.yyy MX]: Remote has diverged from TF Config. Manual intervention required.

### Expected Behavior
terraform apply without error

### Actual Behavior
The error occurs

### Steps to Reproduce
1. `terraform plan`

**Comment by edglynes** [comment]
The provider has correctly identified a mismatch between the existing record in Edgegrid and the local TF config. Because of the nature of MX record configuration options, it cannot be safely reconciled. Specifically, 

"10 mx1.xxxx.yyy.eu.","10 mx2.xxxx.yyy." vs
  target = [
    "mx1."xxx.yyy"",
    "mx2."xxx.yyy""
  ]

There are several differences; embededed quotes and also the mx1 target name.

See https://www.terraform.io/docs/providers/akamai/g/get_started_dns_zone.html#working-with-mx-records for more detailed information.

**Comment by ericrichtert** [comment]
Well, I made some typos in hiding the real names.

But even when I delete _resource "akamai_dns_record" "mx__xxxx_yyy"_ from the tf files, the same error occurs.
And the current config is applied by the TF provider (0.5.0).

**Comment by edglynes** [comment]
I need to see the EXACT configuration to reproduce.  You can send to me directly if thats better, elynes@akamai.com

Did you read the mx-record section in the Getting Started? Did it provide any clarity?

**Comment by edglynes** [comment]
Eric - There should be another log line just above what you included above. It looks like 

[DEBUG] [Akamai DNSv2] READ record JSON from bind records  ...

Can you provide that? Thx.

**Comment by edglynes** [comment]
Confirmed bug. A computed field to track the record's current state, record_sha, was introduced in 0.7.0.  It is used to compare local TF resource state with remote. The issue is that for upgrades with existing state, that field defaults to zero.  Fix will be in the next release.

A workaround is to remove the resource from the current state and then do an import of the resource. 

Corrective steps:
terraform init
terraform rm akamai_dns_record.mx__xxxx_yyy
terraform import akamai_dns_record.mx__xxxx_yyy xxx.yyy#xxx.yyy#MX

**Comment by ericrichtert** [comment]
Good to hear to fixed this issue. When will the new version be released? I have 200+ domains to fix, so a fixed provider is easier for me

**Comment by edglynes** [comment]
Should be this week. Maybe in the next day or two. Release process has already started.  What other record types do you have in your domains? They all behaved appropriately?

Thanks for taking the time to report that issue. We really appreciate it!

Ed

Sent from my iPad

On Jun 1, 2020, at 3:53 AM, ericrichtert <notifications@github.com> wrote:

﻿

Good to hear to fixed this issue. When will the new version be released? I have 200+ domains to fix, so a fixed provider is easier for me

**Comment by ericrichtert** [comment]
@edglynes 
We found two other things:
1) "end_user_id" was filled at initial setup via de webui of akamai. The TF plan command showed the removal, so we added it to the akamai_dns_zone. I think the "end_user_id" is included in version 0.7.0.

2) AAAA records were in the short notation in our tf files. Let say "`1a00:d10:1234:5::22`". That gives an never ending change to "`1a00:0d10:1234:0005:0000:0000:0000:0022`", so we have put the long version in the tf file.

When these tf files are merged in our repo, we will start converting CNAME to AKAMAICDN records for our domains.
